# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: toolkit/modules/SubDialog.sys.mjs
# Commit: beeda5ef3712
# Full Hash: beeda5ef3712b73a19c7a4ce08a52ab4dd08b802
# Author: David P <daparks@mozilla.com>
# Date: 2024-07-04 16:09:52
# Regressor Bug: 1893119
# File Overlap Count: 1
# Description:
#   Bug 1893119: Part 20 - Move EndDragSession from nsIDragService to nsIDragSession r=gstoll,rkraesig,win-reviewers,geckoview-reviewers,m_kato
#   
#   This is a straightforward move.  It does take the liberty of breaking out an
#   EndDragSessionImpl method, which will be needed later.
#   
# ==============================================================================

diff -r 6c3ffc8c534c -r beeda5ef3712 toolkit/modules/SubDialog.sys.mjs
--- a/toolkit/modules/SubDialog.sys.mjs	Thu Jul 04 07:48:10 2024 +0000
+++ b/toolkit/modules/SubDialog.sys.mjs	Thu Jul 04 07:48:11 2024 +0000
@@ -184,8 +184,9 @@
       // The drag service getService call fails in puppeteer tests on Linux,
       // so this is in a try...catch as it shouldn't stop us from opening the
       // dialog. Bug 1806870 tracks fixing this.
-      if (lazy.dragService.getCurrentSession(this._window)) {
-        lazy.dragService.endDragSession(true);
+      let session = lazy.dragService.getCurrentSession(this._window);
+      if (session) {
+        session.endDragSession(true);
       }
     } catch (ex) {
       console.error(ex);